seed/ruby-sdk/oauth-client-credentials-environment-variables/lib/fern_oauth_client_credentials_environment_variables.rb

```diff
@@ -12,14 +13,24 @@ class Client
     # @param base_url [String]
     # @param max_retries [Long] The number of times to retry a failed request, defaults to 2.
     # @param timeout_in_seconds [Long]
-    # @param token [String]
+    # @param client_id [String]
+    # @param client_secret [String]
     # @return [SeedOauthClientCredentialsEnvironmentVariablesClient::Client]
-    def initialize(token:, base_url: nil, max_retries: nil, timeout_in_seconds: nil)
+    def initialize(base_url: nil, max_retries: nil, timeout_in_seconds: nil, client_id: nil, client_secret: nil)
+      @oauth_provider = SeedOauthClientCredentialsEnvironmentVariablesClient::OauthTokenProvider.new(
+        client_id: client_id,
+        client_secret: client_secret,
```

Comment:
Does this support reading the variables from the environment yet? Totally fine if we leave that for a follow-up if not.

Reply:
ah yes forgot to add, will do that now

Reply:
added!